chapters/memory-layout/c-assembly-interaction/drills/tasks/regs-preserve/README.md

```diff
@@ -0,0 +1,22 @@
+# Keeping Records
+
+Navigate to `drills/tasks/regs-preserve/support` and open `main.asm`
+
+In this subdirectory of the lab's task repository you will find the `print_reverse_array()` function implemented by a simple loop that makes repeated calls of the `printf()` function.
+
+Follow the code in the `main.asm` file, compile and run the program. What happened?
+The program runs indefinitely. This is because the `printf()` function does not preserve the value in the `ecx` register, used here as a counter.
```

Comment:
```suggestion
The program runs indefinitely.
This is because the `printf()` function does not preserve the value in the `ecx` register, used here as a counter.
```